### DISCUSSION TITLE
issue: undefined values when deleting items when using useController

### DISCUSSION THREAD

**Discussion by unHuesio** [react-hook-form/react-hook-form#7277] [Q&A]
### Version Number

7.21.2

### Codesandbox/Expo snack

https://codesandbox.io/s/determined-joana-udp02?file=/src/App.js:2848-2866

### Steps to reproduce

1. Click on the add rule button several times and add at least 3 or 4 items
2. Start deleting rules starting from the bottom one
3. The app breaks because of undefined values

### Expected behaviour

The issue occurs specifically when using the useController and setting shouldUnregister to true
Also its combined issue while using framer-motion library to add custom animations

Expected behaviour is that it should be able to delete elements without breaking the application by having undefined values inside the useFieldArray values

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Edge

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
that's an external dependency there, which is not something we support out of the box.

**Reply by unHuesio** [reply]
So nothing to be considered in the future?, otherwise I'll just need to create animations without using the library

**Reply by bluebill1049** [reply]
hey @abFx I am not an expert on framer-motion, at the moment I think it would be difficult by the look with it, probably using native CSS animation would be an easier solution. However, I do understand there will be limitations with unmounting animation.

**Reply by bluebill1049** [reply]
maybe try this as well

```tsx
<AnimatePresence>
  {rules.fields.filter(Boolean).map((rule, index) => {
    return (
      <div key={rule?._id} >
      <motion.li{...motionProps}>
        <EditableRule
          path={`${path}.rules.${index}`}
          rule={rule}
          remove={rules.remove}
          removeFromIndex={index}
        />
        <button
          onClick={() => {
            addRule(index + 1);
          }}
        >
          add rule
        </button>
      </motion.li>
      </div>
    );
  })}
</AnimatePresence>
```

**Reply by bluebill1049** [reply]
https://codesandbox.io/s/cranky-mayer-0emek

**Reply by unHuesio** [reply]
@bluebill1049 thanks for the suggestion!
Though i also tried several filters and patches, but at the end if we submit the form it contains undefined values, you can check the console.log with the fields value
Not an easy workaround at the end
